components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1263,31 +1296,237 @@ public static String getTenantDomainOfOauthApp(OAuthAppDO oAuthAppDO) {
      * @return
      * @throws IdentityOAuth2Exception
      */
-    public static String mapSignatureAlgorithm(String signatureAlgorithm)
-            throws IdentityOAuth2Exception {
-        if ("SHA256withRSA".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.RS256.getName();
-        } else if ("SHA384withRSA".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.RS384.getName();
-        } else if ("SHA512withRSA".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.RS512.getName();
-        } else if ("SHA256withHMAC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.HS256.getName();
-        } else if ("SHA384withHMAC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.HS384.getName();
-        } else if ("SHA512withHMAC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.HS512.getName();
-        } else if ("SHA256withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES256.getName();
-        } else if ("SHA384withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES384.getName();
-        } else if ("SHA512withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES512.getName();
-        } else if (ALGORITHM_NONE.equals(signatureAlgorithm)) {
-            return JWSAlgorithm.NONE.getName();
+    public static JWSAlgorithm mapSignatureAlgorithm(String signatureAlgorithm) throws IdentityOAuth2Exception {
+
+        if (NONE.equalsIgnoreCase(signatureAlgorithm)) {
+            return new JWSAlgorithm(JWSAlgorithm.NONE.getName());
+        } else if (SHA256_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS256;
+        } else if (SHA384_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS384;
+        } else if (SHA512_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS512;
+        } else if (SHA256_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS256;
+        } else if (SHA384_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS384;
+        } else if (SHA512_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS512;
+        } else if (SHA256_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES256;
+        } else if (SHA384_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES384;
+        } else if (SHA512_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES512;
         }
         log.error("Unsupported Signature Algorithm in identity.xml");
         throw new IdentityOAuth2Exception("Unsupported Signature Algorithm in identity.xml");
     }
 
+    /**
+     * This method maps signature algorithm define in identity.xml to digest algorithms to generate the at_hash
```

Comment:
This comment is bit ambiguous . It's not the digest algorithm, isn't it. It is the signature algorithm.